### PR TITLE
[bitnami/mongodb-sharded] Fix uncapacity to disable persistence

### LINKED ISSUE #29127: [mongodb-sharded] Disabling persistence is impossible
### Name and Version

bitnami/mongodb-sharded 8.3.6

### What architecture are you using?

amd64

### What steps will reproduce the bug?

1. `helm install my-mongodb-sharded oci://registry-1.docker.io/bitnamicharts/mongodb-sharded --version=8.2.3 --set shardsvr.persistence.enabled=false`

or

1. `helm install my-mongodb-sharded oci://registry-1.docker.io/bitnamicharts/mongodb-sharded --version=8.2.3 --set configsvr.persistence.enabled=false`

### Are you using any custom parameters or values?

`--set shardsvr.persistence.enabled=false` or `--set configsvr.persistence.enabled=false`

### What is the expected behavior?

My release gets deployed without producing any PV or PVC for the associated MongoDB entity.

### What do you see instead?

`Error: INSTALLATION FAILED: YAML parse error on mongodb-sharded/templates/shard/shard-data-statefulset.yaml: error converting YAML to JSON: yaml: line 184: did not find expected key`

or

`Error: INSTALLATION FAILED: YAML parse error on mongodb-sharded/templates/config-server/config-server-statefulset.yaml: error converting YAML to JSON: yaml: line 187: did not find expected key`

### Additional information

Providing `"false"` as a string produces the same error.

### PR DESCRIPTION
**Pull request by tschneider-aneo**
### Description of the change

This PR modifies how the `datadir` volume is created and mounted when `persistence.enabled` parameter is set to false within statefulset manifests for both config server and shard server. It also removes some bad lines of code.
### Benefits

Persistence can now be disabled for both config server and shard server without resulting in a YAML parse error.
### Applicable issues

- fixes #29127 

### Additional information

Disabling persistence can be useful when the default configuration to be set doesn't fit your cluster.
### Checklist

- [X ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X ] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X ] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X ] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/mongodb-sharded/templates/shard/shard-data-statefulset.yaml [hunk 1]
```diff
@@ -428,7 +428,10 @@ spec:
         {{- if $.Values.shardsvr.dataNode.extraVolumes }}
           {{- include "common.tplvalues.render" ( dict "value" $.Values.shardsvr.dataNode.extraVolumes "context" $ ) | nindent 8 }}
         {{- end }}
-  {{- if $.Values.shardsvr.persistence.enabled }}
+  {{if not $.Values.shardsvr.persistence.enabled}}
```
Comment (andresbono) [inline bitnami/mongodb-sharded/templates/shard/shard-data-statefulset.yaml:431]: Small suggestion

```suggestion
  {{- if not $.Values.shardsvr.persistence.enabled }}
```

**Review by andresbono — COMMENTED** [review]
Thank you for contributing with this fix! So it was a matter of wrong indentation in the end, right?

Please, check my small suggestion.

**Comment by tschneider-aneo** [comment]
Hi @andresbono, thanks for your review and suggestion ! I have just applied it !
It was indeed mostly an indentation error.

**Review by andresbono — APPROVED** [review]
LGTM, thank you for your contribution!

**Comment by tschneider-aneo** [comment]
Following this PR, I see in the ArtifactHub that versions of the chart goes from 8.3.6 to 8.3.8, skipping 8.3.7. . I changed the version to 8.3.8 because the previous PR created a 8.3.7. but it doesn't seem to have been published. Is this a problem for you ?

**Comment by carrodher** [comment]
I just fetched the latest version and it contains those changes, could you please confirm if something else is missing?

**Pull request merged**